modules/mono/glue/GodotSharp/GodotPlugins/PluginLoadContext.cs

```diff
@@ -21,6 +21,20 @@ public class PluginLoadContext : AssemblyLoadContext
             _resolver = new AssemblyDependencyResolver(pluginPath);
             _sharedAssemblies = sharedAssemblies;
             _mainLoadContext = mainLoadContext;
+
+            if (string.IsNullOrEmpty(AppContext.BaseDirectory))
+            {
+                // See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs#L18-L36
+                // but Assembly.Location is unavailable, because we load assemblies from memory.
+                string? baseDirectory = Path.GetDirectoryName(pluginPath);
+                if (baseDirectory != null)
```

Comment:
Let's print an error when this condition is false, to have context if a library cannot be loaded afterwards.

Reply:
I added an error, tho the else branch is effectively dead code as `pluginPath` always has a directory name and thus GetDirectoryName never actually returns null. (I only really added the if, because otherwise the compiler complains and the runtime also has it.)